pkg/controller/remoteingress/remoteingress_controller.go

```diff
@@ -0,0 +1,234 @@
+		cdLog.Errorf("error syncing clusterIngress syncset: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileRemoteClusterIngress) syncClusterIngress(cd *hivev1.ClusterDeployment) error {
+	cdLog := r.logger.WithFields(log.Fields{
+		"clusterDeployment": cd.Name,
+		"namespace":         cd.Namespace,
+	})
+	cdLog.Info("reconciling ClusterIngress for cluster deployment")
+
+	rawList := rawExtensionsFromClusterDeployment(cd)
+
+	if err := r.syncSyncSet(cd, rawList); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func rawExtensionsFromClusterDeployment(cd *hivev1.ClusterDeployment) []runtime.RawExtension {
+	rawList := []runtime.RawExtension{}
+	for _, ingress := range cd.Spec.Ingress {
+		ingressObj := createClusterIngress(cd, ingress)
+		raw := runtime.RawExtension{Object: ingressObj}
+		rawList = append(rawList, raw)
+	}
+
+	return rawList
+}
+
+func newSyncSetSpec(cd *hivev1.ClusterDeployment, rawExtensions []runtime.RawExtension) *hivev1.SyncSetSpec {
+	ssSpec := &hivev1.SyncSetSpec{
+		SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+			Resources:         rawExtensions,
+			ResourceApplyMode: "sync",
+		},
+		ClusterDeploymentRefs: []corev1.LocalObjectReference{
+			{
+				Name: cd.Name,
+			},
+		},
+	}
+	return ssSpec
+}
+
+func (r *ReconcileRemoteClusterIngress) syncSyncSet(cd *hivev1.ClusterDeployment, rawExtensions []runtime.RawExtension) error {
+
+	ssName := fmt.Sprintf("%v-clusteringress", cd.Name)
+	newSyncSetSpec := newSyncSetSpec(cd, rawExtensions)
+
+	ss := &hivev1.SyncSet{}
+	err := r.Get(context.TODO(), types.NamespacedName{Name: ssName, Namespace: cd.Namespace}, ss)
+	if errors.IsNotFound(err) {
+		// create the syncset
+		ss = &hivev1.SyncSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ssName,
+				Namespace: cd.Namespace,
+			},
+			Spec: *newSyncSetSpec,
+		}
+
+		// ensure the syncset gets cleaned up when the clusterdeployment is deleted
+		controllerutil.SetControllerReference(cd, ss, r.scheme)
+
+		if err := r.Create(context.TODO(), ss); err != nil {
+			r.logger.Errorf("error creating sync set: %v", err)
+			return err
+		}
+
+		return nil
+	} else if err != nil {
+		r.logger.Errorf("error checking for existing syncset: %v", err)
+		return err
+	}
+
+	// update the syncset if there have been changes
+	if !reflect.DeepEqual(ss.Spec, *newSyncSetSpec) {
+		ss.Spec = *newSyncSetSpec
+		if err := r.Update(context.TODO(), ss); err != nil {
+			errDetails := fmt.Errorf("error updating existing syncset: %v", err)
+			r.logger.Error(errDetails)
+			return errDetails
+		}
+	}
+
+	return nil
+}
+
+func createClusterIngress(cd *hivev1.ClusterDeployment, ingress hivev1.ClusterIngress) *clusteringress.ClusterIngress {
+	newIngress := clusteringress.ClusterIngress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingress.Name,
+			Namespace: remoteClusterIngressNamespace,
+		},
+		Spec: clusteringress.ClusterIngressSpec{
+			IngressDomain: stringPtr(ingress.IngressDomain),
```

Comment:
@joelddiaz should we have defaults for settings like replicas and nodePlacement?

Reply:
the new definitions in openshift/api have defaults for replicas (2), and there is a default (but not defined in the comments) for nodeplacement https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L114-L120